src/audit/template_injection.rs

```diff
@@ -87,40 +89,57 @@ const SAFE_CONTEXTS: &[&str] = &[
 ];
 
 impl TemplateInjection {
-    fn script_with_location<'s>(
+    /// For the name of an action (e.g. `actions/github-script`) gets the `inputs`
+    /// which may allow code injection.
+    fn action_injection_inputs(action_name: &str) -> &[&'static str] {
+        static MODEL_INJECTION_SINKS: LazyLock<HashMap<&str, Vec<&str>>> = LazyLock::new(|| {
+            let mut m = HashMap::new();
+            let model_data = include_str!("../../code-injection-models.txt");
+
+            for line in model_data.split_terminator("\n") {
+                let (action, inputs) = line.split("|").collect_tuple().unwrap();
+                let inputs = inputs.split(",").collect();
+                m.insert(action, inputs);
+            }
+
+            m
+        });
+
+        match action_name {
+            // Hardcoded sinks which are not part of CodeQL models yet
+            "amadevus/pwsh-script" => &["script"],
+            "jannekem/run-python-script-action" => &["script"],
+            "cardinalby/js-eval-action" => &["expression"],
+            _ => match MODEL_INJECTION_SINKS.get(action_name) {
+                Some(inputs) => inputs,
+                None => &[],
+            },
+        }
+    }
+
+    fn scripts_with_location<'s>(
         step: &impl StepCommon<'s>,
-    ) -> Option<(String, SymbolicLocation<'s>)> {
+    ) -> Vec<(String, SymbolicLocation<'s>)> {
```

Comment:
Changed to `Vec` because this can in theory now return multiple vulnerable `inputs`.